File: connectors/hana-hook/shared/work-receipt.js
const MARKERS = ["🐳 十元在线", "🐳 公司十元本地模式"];
const EVIDENCE = ["通过", "passed", "healthy", "测试", "验证", "sha-256", "commit", "提交", "校验"];
const NEXT = ["下一步", "未完成", "尚未", "还需", "仍需", "待处理", "遗留", "需要重启"];
const DECISION = ["决定", "采用", "选择", "保持", "不再", "改为", "范围"];

function matchedLines(lines, cues, limit = 5) {
  const values = [];
  for (const line of lines) {
    const lower = line.toLowerCase();
    if (cues.some((cue) => lower.includes(cue))) {
      const value = line.replace(/^[\s#>*+\-\d.）)✅⚠️🔍]+/u, "").trim().slice(0, 500);
      if (value && !values.includes(value)) values.push(value);
    }
    if (values.length >= limit) break;
  }
  return values;
}

export function messageText(message) {
  if (typeof message?.content === "string") {
    return message.content.replace(/<pulse\b[^>]*>[\s\S]*?<\/m?pulse>\s*/giu, "").trim();
  }
  if (!Array.isArray(message?.content)) return "";
  return message.content
    .filter((part) => part?.type === "text")
    .map((part) => part.text || "")
    .join("\n")
    .replace(/<pulse\b[^>]*>[\s\S]*?<\/m?pulse>\s*/giu, "")
    .trim();
}

export function compactAssistantMessage(message) {
  let text = String(message || "").replace(/\r\n/g, "\n");
  for (const marker of MARKERS) text = text.replaceAll(marker, "");
  text = text.trim();
  const lines = text.split("\n").map((line) => line.trim()).filter(Boolean);
  const paragraphs = text.split(/\n\s*\n/).map((part) => part.trim()).filter(Boolean);
  const selected = [];
  for (const paragraph of paragraphs) {
    if ((paragraph.startsWith("#") || paragraph.startsWith("```")) && selected.length === 0) continue;
    selected.push(paragraph);
    if (selected.join("\n\n").length >= 900) break;
  }
  const resultSummary = selected.join("\n\n").trim().slice(0, 900)
    || "本轮没有可保存的用户可见结果摘要。";
  const artifacts = [];
  const patterns = [/\[[^\]]+\]\(([^)]+)\)/g, /`([^`\n]*(?:[A-Za-z]:\\|\/)[^`\n]+)`/g, /(?<!\w)([A-Za-z]:\\[^\s<>"|?*]+)/g];
  for (const pattern of patterns) {
    for (const match of text.matchAll(pattern)) {
      const value = match[1]?.trim();
      if (value && !value.startsWith("http://") && !value.startsWith("https://") && !artifacts.includes(value)) {
        artifacts.push(value.slice(0, 1000));
      }
      if (artifacts.length >= 12) break;
    }
  }
  const nextActions = matchedLines(lines, NEXT);
  const lower = text.toLowerCase();
  const blocked = ["阻塞", "无法继续", "需要用户", "等待授权", "失败"].some((cue) => lower.includes(cue));
  const completed = ["已完成", "完成了", "全部通过", "已经解决", "已部署"].some((cue) => lower.includes(cue));
  return {
    status: blocked ? "blocked" : completed && nextActions.length === 0 ? "completed" : "waiting",
    result_summary: resultSummary,
    decisions: matchedLines(lines, DECISION),
    artifacts,
    evidence: matchedLines(lines, EVIDENCE),
    next_actions: nextActions,
  };
}

File: connectors/codex-hook/work_receipt.py
from __future__ import annotations

import re


MARKERS = ("🐳 十元在线", "🐳 公司十元本地模式")
EVIDENCE_CUES = ("通过", "passed", "healthy", "测试", "验证", "sha-256", "commit", "提交", "校验")
NEXT_CUES = ("下一步", "未完成", "尚未", "还需", "仍需", "待处理", "遗留", "需要重启")
DECISION_CUES = ("决定", "采用", "选择", "保持", "不再", "改为", "范围")


def _matches(lines: list[str], cues: tuple[str, ...], limit: int = 5) -> list[str]:
    result = []
    for line in lines:
        if any(cue in line.lower() for cue in cues):
            value = re.sub(r"^[\s#>*+\-\d.）)✅⚠️🔍]+", "", line).strip()
            if value and value not in result:
                result.append(value[:500])
        if len(result) >= limit:
            break
    return result


def compact_assistant_message(message: str) -> dict:
    text = str(message or "").replace("\r\n", "\n")
    for marker in MARKERS:
        text = text.replace(marker, "")
    text = text.strip()
    lines = [line.strip() for line in text.splitlines() if line.strip()]
    paragraphs = [part.strip() for part in re.split(r"\n\s*\n", text) if part.strip()]
    chosen = []
    for part in paragraphs:
        if part.startswith(("#", "```")) and not chosen:
            continue
        chosen.append(part)
        if len("\n\n".join(chosen)) >= 900:
            break
    summary = "\n\n".join(chosen).strip()[:900] or "本轮没有可保存的用户可见结果摘要。"
    artifacts = []
    found = re.findall(r"\[[^\]]+\]\(([^)]+)\)", text)
    found += re.findall(r"`([^`\n]*(?:[A-Za-z]:\\|/)[^`\n]+)`", text)
    found += re.findall(r"(?<!\w)([A-Za-z]:\\[^\s<>\"|?*]+)", text)
    for value in found:
        value = value.strip()
        if value.startswith(("http://", "https://")) or not value:
            continue
        if value not in artifacts:
            artifacts.append(value[:1000])
        if len(artifacts) >= 12:
            break
    next_actions = _matches(lines, NEXT_CUES)
    lowered = text.lower()
    blocked = any(cue in lowered for cue in ("阻塞", "无法继续", "需要用户", "等待授权", "失败"))
    completed = any(cue in lowered for cue in ("已完成", "完成了", "全部通过", "已经解决", "已部署"))
    return {
        "status": "blocked" if blocked else "completed" if completed and not next_actions else "waiting",
        "result_summary": summary,
        "decisions": _matches(lines, DECISION_CUES),
        "artifacts": artifacts,
        "evidence": _matches(lines, EVIDENCE_CUES),
        "next_actions": next_actions,
    }
